scripts/Diffraction/isis_powder/routines/focus.py

```diff
@@ -48,22 +48,37 @@ def _focus_one_ws(input_workspace, run_number, instrument, perform_vanadium_norm
         summed_empty = common.generate_summed_runs(empty_sample_ws_string=run_details.sample_empty,
                                                    instrument=instrument,
                                                    scale_factor=instrument._inst_settings.sample_empty_scale)
-    if summed_empty is not None:
-        input_workspace = common.subtract_summed_runs(ws_to_correct=input_workspace,
-                                                      empty_sample=summed_empty)
 
-    # Crop to largest acceptable TOF range
-    input_workspace = instrument._crop_raw_to_expected_tof_range(ws_to_crop=input_workspace)
-
-    # Correct for absorption / multiple scattering if required
     if absorb:
-        input_workspace = instrument._apply_absorb_corrections(run_details=run_details, ws_to_correct=input_workspace)
+        if summed_empty:
```

Comment:
Couple of things about this if statement:

1) I think the condition for running PP should instead be `if run_details.sample_empty:`
This relates to the conversation we were having with Helen about the two types of "empty" run. If summed_empty contains the empty instrument run (ie no sample and no can) then we shouldn't be doing the PP approach. This potentially complicates the if\else statements over the next ~20 lines...

2) the rework of the logic over the next ~20 lines means that the route through the code for people not running the PP approach will change a bit. Most of it should be functionally equivalent - although the order of the cropping\absorption correction steps have swapped - but even so it introduces a bit of risk that we may break something. Could we do something a bit more like the following instead? As well as keeping the non-PP code closer to the original, it has a bit less duplication in it I think:
```
if absorb and PP:
  if run_details.sample_empty:
    instrument._apply_paalmanpings_absorb_and_subtract_empty
  else:
    error
else
  if summed_empty:
    common.subtract_summed_runs
  crop_raw_to_expected_tof_range
  if absorb:
    instrument._apply_absorb_corrections
```